base/applications/charmap/map.c

```diff
@@ -312,10 +351,25 @@ OnClick(PMAP infoPtr,
         WORD ptx,
         WORD pty)
 {
-    INT x, y;
-
-    x = ptx / max(1, infoPtr->CellSize.cx);
-    y = pty / max(1, infoPtr->CellSize.cy);
+    INT x, y, i;
+
+    /* find the cell the mouse pointer is over.
+     * since each cell is the same size this can be done quickly using CellSize
+     * clamp to XCELLS - 1 and YCELLS - 1 because the map can sometimes be slightly
+     * larger than infoPtr.CellSize * XCELLS due to the map size being a non integer 
+     * multiple of infoPtr.CellSize */
```

Comment:
```suggestion
    /*
     * Find the cell the mouse pointer is over.
     * Since each cell is the same size, this can be done quickly using CellSize.
     * Clamp to XCELLS - 1 and YCELLS - 1 because the map can sometimes be slightly
     * larger than infoPtr.CellSize * XCELLS , due to the map size being a non integer 
     * multiple of infoPtr.CellSize .
     */
```